extension/intents/notes/notes.js

```diff
@@ -84,3 +84,19 @@ intentRunner.registerIntent({
     await context.makeTabActive(writingTabId);
   },
 });
+
+intentRunner.registerIntent({
+  name: "notes.paste",
+  async run(context) {
+    await checkHasTab();
+    const success = await browser.tabs.sendMessage(writingTabId, {
+      type: "addText",
+      text: document.execCommand("paste"),
```

Comment:
`document.execCommand("paste")` doesn't return anything useful (it may return true/false for success/failure, but even that seems unreliable). Instead it actually performs the paste.

So to get this to paste in a document you'll need to add to `extension/intents/notes/contentScript.js` – a new `communicate.register("paste", () => {...})`

That command should call `focus()` and then `document.execCommand("paste")`. It could return the value of document.execCommand, but it's very possible that's going to give false errors to the user.

Reply:
Thanks! I'll make the changes

Reply:
communicate.register("paste", () => {...}) is similar to that of addText right?

Reply:
Yep!